packages/expo-notifications/android/src/main/java/expo/modules/notifications/notifications/service/ExpoNotificationSchedulerService.java

```diff
@@ -270,7 +270,7 @@ protected NotificationRequest fetchNotification(String identifier) {
   }
 
   /**
-   * Handle notification to schedule. If trigger is null, hands the request immediately to {@link BaseNotificationsService}.
+   * Handle notification to schedule. If trigger is null or isn't {@link SchedulableNotificationTrigger}, hands the request immediately to {@link BaseNotificationsService}.
```

Comment:
Can't we handle non schedulable triggers just by checking getNextTriggerDate here, instead of passing it? That would simplify our hierarchy, wouldn't it?

Reply:
We can, but it will be less efficient - if `getNextTriggerDate` returns null, we need to delete the notification request. So, if we check it here, we need to try to delete every request, even if it wasn't scheduled.